File: SDKs/nexmo.com/number-insight/1.2.1/python/src/sdk/models/shared/niresponsejsonstandard.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *

class NiResponseJSONStandardCallerTypeEnum(str, Enum):
    BUSINESS = "business"
    CONSUMER = "consumer"
    UNKNOWN = "unknown"

class NiResponseJSONStandardCurrentCarrierNetworkTypeEnum(str, Enum):
    MOBILE = "mobile"
    LANDLINE = "landline"
    LANDLINE_PREMIUM = "landline_premium"
    LANDLINE_TOLLFREE = "landline_tollfree"
    VIRTUAL = "virtual"
    UNKNOWN = "unknown"
    PAGER = "pager"
    NULL = "null"


@dataclass_json
@dataclass
class NiResponseJSONStandardCurrentCarrier:
    country: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('country') }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    network_code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('network_code') }})
    network_type: Optional[NiResponseJSONStandardCurrentCarrierNetworkTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('network_type') }})
    
class NiResponseJSONStandardPortedEnum(str, Enum):
    UNKNOWN = "unknown"
    PORTED = "ported"
    NOT_PORTED = "not_ported"
    ASSUMED_NOT_PORTED = "assumed_not_ported"
    ASSUMED_PORTED = "assumed_ported"
    NULL = "null"


@dataclass_json
@dataclass
class NiResponseJSONStandard:
    r"""NiResponseJSONStandard
    Standard Response
    """
    
    caller_identity: Optional[NiCallerIdentity] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('caller_identity') }})
    caller_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('caller_name') }})
    caller_type: Optional[NiResponseJSONStandardCallerTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('caller_type') }})
    country_code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('country_code') }})
    country_code_iso3: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('country_code_iso3') }})
    country_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('country_name') }})
    country_prefix: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('country_prefix') }})
    current_carrier: Optional[NiResponseJSONStandardCurrentCarrier] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current_carrier') }})
    first_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('first_name') }})
    international_format_number: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('international_format_number') }})
    last_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('last_name') }})
    national_format_number: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('national_format_number') }})
    original_carrier: Optional[NiInitialCarrierProperties] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('original_carrier') }})
    ported: Optional[NiResponseJSONStandardPortedEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ported') }})
    refund_price: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('refund_price') }})
    remaining_balance: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('remaining_balance') }})
    request_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('request_id') }})
    request_price: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('request_price') }})
    status: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    status_message: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status_message') }})
    

File: SDKs/nexmo.com/number-insight/1.2.1/typescript/src/sdk/models/shared/niresponsejsonstandard.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { NiCallerIdentity } from "./nicalleridentity";
import { NiInitialCarrierProperties } from "./niinitialcarrierproperties";


export enum NiResponseJsonStandardCallerTypeEnum {
    Business = "business",
    Consumer = "consumer",
    Unknown = "unknown"
}

export enum NiResponseJsonStandardCurrentCarrierNetworkTypeEnum {
    Mobile = "mobile",
    Landline = "landline",
    LandlinePremium = "landline_premium",
    LandlineTollfree = "landline_tollfree",
    Virtual = "virtual",
    Unknown = "unknown",
    Pager = "pager",
    Null = "null"
}


export class NiResponseJsonStandardCurrentCarrier extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=country" })
  country?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=network_code" })
  networkCode?: string;

  @SpeakeasyMetadata({ data: "json, name=network_type" })
  networkType?: NiResponseJsonStandardCurrentCarrierNetworkTypeEnum;
}

export enum NiResponseJsonStandardPortedEnum {
    Unknown = "unknown",
    Ported = "ported",
    NotPorted = "not_ported",
    AssumedNotPorted = "assumed_not_ported",
    AssumedPorted = "assumed_ported",
    Null = "null"
}


// NiResponseJsonStandard
/** 
 * Standard Response
**/
export class NiResponseJsonStandard extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=caller_identity" })
  callerIdentity?: NiCallerIdentity;

  @SpeakeasyMetadata({ data: "json, name=caller_name" })
  callerName?: string;

  @SpeakeasyMetadata({ data: "json, name=caller_type" })
  callerType?: NiResponseJsonStandardCallerTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=country_code" })
  countryCode?: string;

  @SpeakeasyMetadata({ data: "json, name=country_code_iso3" })
  countryCodeIso3?: string;

  @SpeakeasyMetadata({ data: "json, name=country_name" })
  countryName?: string;

  @SpeakeasyMetadata({ data: "json, name=country_prefix" })
  countryPrefix?: string;

  @SpeakeasyMetadata({ data: "json, name=current_carrier" })
  currentCarrier?: NiResponseJsonStandardCurrentCarrier;

  @SpeakeasyMetadata({ data: "json, name=first_name" })
  firstName?: string;

  @SpeakeasyMetadata({ data: "json, name=international_format_number" })
  internationalFormatNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=last_name" })
  lastName?: string;

  @SpeakeasyMetadata({ data: "json, name=national_format_number" })
  nationalFormatNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=original_carrier" })
  originalCarrier?: NiInitialCarrierProperties;

  @SpeakeasyMetadata({ data: "json, name=ported" })
  ported?: NiResponseJsonStandardPortedEnum;

  @SpeakeasyMetadata({ data: "json, name=refund_price" })
  refundPrice?: string;

  @SpeakeasyMetadata({ data: "json, name=remaining_balance" })
  remainingBalance?: string;

  @SpeakeasyMetadata({ data: "json, name=request_id" })
  requestId?: string;

  @SpeakeasyMetadata({ data: "json, name=request_price" })
  requestPrice?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: number;

  @SpeakeasyMetadata({ data: "json, name=status_message" })
  statusMessage?: string;
}
